pkg/proc/gdbserial/undo.go

```diff
@@ -156,6 +455,13 @@ func UndoReplay(recording string, path string, quiet bool, debugInfoDirs []strin
 	// to the GDB mapping, not the Linux mapping (the binutils-gdb repo
 	// defines the GDB mapping in include/gdb/signals.def)
 	p.conn.isUndoServer = true
+
+	// Create storage for Undo checkpoints, which (unlike rr) aren't stored in the server.
+	p.undoSession = newUndoSession()
```

Comment:
See note above — `p.conn.isUndoServer` is redundant with `p.undoSession != nil` so maybe the former could now be dropped. (By dropping redundant fields we avoid the risk of the data structure getting into an inconsistent state.)

Reply:
This'll want looking into but I think it's a future thing - the current separation is something we've inherited from the original implementation, it might make sense or not be required at all.